lists/list_cache.go

```diff
@@ -117,152 +109,138 @@ func logger() *logrus.Entry {
 	return log.PrefixedLog("list_cache")
 }
 
-// downloads and reads files with domain names and creates cache for them
-//
-//nolint:funlen // will refactor in a later commit
-func (b *ListCache) createCacheForGroup(group string, links []string) (created bool, err error) {
-	groupFactory := b.groupedCache.Refresh(group)
-
-	fileLinesChan := make(chan string, chanCap)
-	errChan := make(chan error, chanCap)
+// Match matches passed domain name against cached list entries
+func (b *ListCache) Match(domain string, groupsToCheck []string) (groups []string) {
+	return b.groupedCache.Contains(domain, groupsToCheck)
+}
 
-	workerDoneChan := make(chan bool, len(links))
+// Refresh triggers the refresh of a list
+func (b *ListCache) Refresh() {
+	_ = b.refresh()
+}
 
-	// guard channel is used to limit the number of concurrent executions of the function
-	guard := make(chan struct{}, b.processingConcurrency)
+func (b *ListCache) refresh() error {
+	unlimitedGrp, _ := jobgroup.WithContext(context.Background())
+	defer unlimitedGrp.Close()
 
-	processingLinkJobs := len(links)
+	producersGrp := jobgroup.WithMaxConcurrency(unlimitedGrp, b.processingConcurrency)
```

Comment:
This limit applies to the whitelist and blacklist independently.  
With the new code it should be easy to make both share the same limit. That seems like more user friendly behavior to me since there's a single option in the config. Do you think I should make that change?

Reply:
I think, one limit for both is good enough

Reply:
Do you mean the current code is fine?  
TBH it feels like we're not really respecting the user's choice here to me. But it's already the case so if you want to keep it as is, it's fine by me.